server/game/cards/events/04/tyrionschain.js

```diff
@@ -0,0 +1,64 @@
+const _ = require('underscore');
+const DrawCard = require('../../../drawcard.js');
+
+class TyrionsChain extends DrawCard {
+    setupCardAbilities(ability) {
+        this.reaction({
+            when: {
+                afterChallenge: (event, challenge) => (
+                    challenge.winner === this.controller &&
+                    this.hasParticipatingUniqueLannister()
+                )
+            },
+            max: ability.limit.perPhase(1),
+            condition: () => this.game.anyPlotHasTrait('War'),
```

Comment:
I think the condition property doesn't work on reactions/interrupts, just actions. So you can move this check to the event listener for the desired result.